src/PositionManager/PositionManager.cpp

```diff
@@ -52,7 +52,12 @@ void QGCPositionManager::setToolbox(QGCToolbox *toolbox)
 void QGCPositionManager::setNmeaSourceDevice(QIODevice* device)
 {
     if (_nmeaSource) {
+        if (_nmeaSource == _currentSource){
+            _currentSource = nullptr;
```

Comment:
make sure you perform the appropriate cleanup [here](https://github.com/mavlink/qgroundcontrol/pull/7087/commits/45321bfb16406cbc0021604f496eecf41d80533b#diff-dc7e6678e48a96d0ba57eab3b2676189R96)